### PR TITLE
[plugins] Add initial sketch of an llbuild extensions API.

### PR DESCRIPTION
**Pull request by ddunbar**
* This is completely experimental at this time, see docs/ for more info.

 * [x] Depends on #442

### CODE REVIEW

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

File: docs/buildsystem-extensions.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+of projects. The extension system is designed to allow a transparent adoption of
+extension behavior, while also allowing those external projects to be decoupled
+from llbuild (i.e. support for llbuild can be something the project adds as a
+benefit to llbuild users, without the llbuild project needing to be aware of it,
+and without needing a complex system-wide registration process).
+
+### Extension Discovery
+
+llbuild dynamically discovers the presence of an "extension" for a particular
+tool, but searching the file system for an additional extension configuration
```
Comment (dmbryson) [inline docs/buildsystem-extensions.md]: s/, but/by/

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Pull request merged**